x-pack/plugins/screenshotting/server/formats/pdf/index.ts

```diff
@@ -6,6 +6,9 @@
  */
 
 import { groupBy } from 'lodash';
+// FIXME: Once/if we have the ability to get page count directly from Chrome/puppeteer
+// we should get rid of this lib.
+import * as PDFJS from 'pdfjs-dist/legacy/build/pdf.js';
```

Comment:
I could not see another way to reliably get page numbers from the PDF buffer produced by Chromium. The puppeteer API does not allow for this and I could not figure out how to do this with `pdfmake`.

`pdfjs` is 100% JS so fortunately no extra native deps, but it does add a fairly large module for a singular purpose - getting page numbers.

Reply:
I put some time into this as well. I hoped it would be easy to find the page count somehow by parsing a PDF buffer, but no luck.

There may be a way to guess the number of pages, based on the number of visualization containers on the page, but we don't pass that info down to this scope.